### PR TITLE
Shortcodes: Update readfile to throw compile error when file not found

### PR DESCRIPTION
**Pull request by aimeeu**
* Update the `readfile` shortcode to throw a compile error when the file is not found. 
* Update userguide

Currently the shortcode prints a message on the UI, which presents a problem when refactoring a docs site. A UI message requires visual conformation and is easy to miss when moving or renaming directories. Throwing a compile error ensures the tech writer knows there is a problem.

With this modification, the default behavior is to throw a compile error:

readfile-compile-error

Alternately, the user can display a message on the page instead of having Hugo throw a compile error (this implements deining's suggestion)

readfile-ui-msg

### CODE REVIEW

**Comment by aimeeu** [comment]
@LisaFC can someone review this please?

**Comment by LisaFC** [comment]
Looks fine to me, but @geriom can you take a look as you wrote the original shortcode?

**Review by LisaFC — APPROVED** [review]

**Review by deining — COMMENTED** [review]
```
A UI message requires visual conformation and is easy to miss when moving or renaming directories.
```

I see the point, and yes, I consider this PR as an improvement and agree that throwing a compile error should be made the default behaviour. However, I can imagine situations where throwing a compile error is unwanted (e.g. with draft documents). What about adding and  evaluating a shortcode parameter `draft`? With attribute draft set to `true`, the current behaviour (printing out an error message in the code) would apply. This way, one could author:

{{< readfile draft=true >}}
...
{{< /readfile >}}

Without parameter draft, a compile error would be thrown.

Just my tow cents ....

File: layouts/shortcodes/readfile.html [hunk 1]
```diff
@@ -34,8 +34,4 @@
   {{ else }}
     {{- $.Scratch.Get "filepath" | os.ReadFile | .Page.RenderString | safeHTML -}}
   {{ end }}
-{{ else }}
-
-<p style="color: #D74848"><b><i>The file <code>{{ $.Scratch.Get "filepath" }}</code> was not found.</i></b></p>
-
-{{ end }}
+{{ else }}{{- errorf "File not found: %s %s" ($.Scratch.Get "filepath") .Position -}}{{ end }}
```
[the remark this answers is not in the document]
Comment (deining) [reply layouts/shortcodes/readfile.html]: Great! LGTM now!

**Review by deining — APPROVED** [review]

**Review by geriom — APPROVED** [review]
LGTM. Thank you!

**Pull request merged**